http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/LeoAppServiceInterp.scala

```diff
@@ -806,6 +796,35 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
       case (_, Some(gcpContext))   => F.pure[CloudContext](CloudContext.Gcp(gcpContext))
       case (None, None) => F.raiseError[CloudContext](CloudContextNotFoundException(workspaceId, ctx.traceId))
     }
+    // check if app can be deleted (Leo manages apps, so checking the leo status)
+    _ <- F.raiseUnless(app.status.isDeletable)(
+      AppCannotBeDeletedException(cloudContext, app.appName, app.status, ctx.traceId)
+    )
+
+    // check if databases, namespaces and managed identities associated with the app can be deleted
+    // (but only for Azure apps)
+    _ <-
+      if (cloudContext.cloudProvider == CloudProvider.Azure) {
+        for {
+          _ <- checkIfSubresourcesDeletable(app.id, WsmResourceType.AzureDatabase, userInfo, workspaceId)
+          _ <- checkIfSubresourcesDeletable(app.id, WsmResourceType.AzureManagedIdentity, userInfo, workspaceId)
+          _ <- checkIfSubresourcesDeletable(app.id, WsmResourceType.AzureKubernetesNamespace, userInfo, workspaceId)
+        } yield ()
+      } else F.unit
```

Comment:
I am curious why we are checking individually if each of these can be deleted before then deleting the disk?
Two questions:
- was this an issue with the pre-existing code?, or
- is this something we can delegate to WSM since they appear to be the owning entity for these resources?

Reply:
These are resources specific to apps (not disks), and before this we weren't checking if they are deletable, just (blindly) sending a delete message and not polling on it. WSM has no concept of these resources being owned by specific apps, that's something Leo manages, so that's why they're checked here. This could really be before or after checking the disk state, they're all sub-resources, but since we don't always delete disks, there's more logic necessary that means its separated out.

Reply:
also disks aren't _technically_ used for Azure apps yet, but with the Rstudio app they will be, so I thought it made sense to include with the rest of this work